File: 4.py
def ispal(n):
    return str(n) == str(n)[::-1]

best = 0
for i in xrange(100,1000):
    for j in xrange(100,1000):
        k = i*j
        if(k > best and ispal(k)):
            best = k
print best

File: js/4.js
reverse = function (s) {
    return s.toString().split("").reverse().join("");
}

ispal = function (x) {
    return x.toString() === reverse(x)
}
var best = 0;
for (var x = 100; x < 1000; x++) {
    for (var y = 100; y < 1000; y++) {
        var product = x * y;
        if (product < best) {
            continue;
        }
        if (ispal(product)) {
            best = product;
        }
    }
}
console.log(best);